src/routes/profiles.py

```
@@ -2,3 +1,127 @@
+        manager: JWTAuthManager = Depends(get_jwt_auth_manager),
+        storage: S3StorageInterface = Depends(get_s3_storage_client),
+        profile_form: ProfileRequestForm = Depends(
+            ProfileRequestForm.as_form
+        ),
+):
+    try:
+        decoded_access_token = manager.decode_access_token(token)
+    except TokenExpiredError:
+        raise HTTPException(status_code=401, detail="Token has expired.")
+
+    try:
+        validate_name(profile_form.first_name)
+        validate_name(profile_form.last_name)
+        validate_birth_date(profile_form.date_of_birth)
+        validate_gender(profile_form.gender)
+
+        if not profile_form.info.strip():
+            raise ValueError(
+                "Info field cannot be empty or contain only spaces."
+            )
+
+        validate_image(profile_form.avatar)
+
+    except ValueError as error:
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail=str(error)
+        )
+
+    user = db.query(UserModel).filter(
+        UserModel.id == decoded_access_token.get("user_id")
+    ).first()
+
+    if not user or user.is_active is False:
+        raise HTTPException(
+            status_code=401,
+            detail="User not found or not active."
+        )
+
+    if user_id != user.id and user.group.name != "admin":
+        raise HTTPException(
+            status_code=403,
+            detail="You don't have permission to edit this profile."
+        )
+
+    profile_exist = db.query(UserProfileModel).filter(
+        UserProfileModel.user_id == user_id
+    ).first()
+
+    if profile_exist:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="User already has a profile."
+        )
+
+    try:
+        filename = f"avatars/{user_id}_avatar.jpg"
+        contents = profile_form.avatar.file.read()
+        storage.upload_file(filename, contents)
+    except Exception:
```

Catching a general `Exception` is not recommended as it can mask other unexpected errors. Consider catching specific exceptions related to file upload failures.